sarif/sarif_test.go

```diff
@@ -210,3 +236,1552 @@ func TestReportFromBytesV2_1_0(t *testing.T) {
+				Metadata: &ocsf.Metadata{
+					EventCode: ptr.Ptr("go.lang.security.audit.dangerous-exec-command.dangerous-exec-command"),
+					Product: &ocsf.Product{
+						Name: ptr.Ptr("Semgrep OSS"),
+					},
+					Labels: []string{"{}"},
+				},
+				Severity:   ptr.Ptr("SEVERITY_ID_MEDIUM"),
+				SeverityId: ocsf.VulnerabilityFinding_SEVERITY_ID_MEDIUM,
+				Time:       now.Unix(),
+				TimeDt:     timestamppb.New(now),
+				TypeName:   ptr.Ptr("Create"),
+				TypeUid:    int64(200201),
+				StartTime:  ptr.Ptr(now.Unix()),
+				StatusId:   ptr.Ptr(ocsf.VulnerabilityFinding_STATUS_ID_NEW),
+				Status:     ptr.Ptr("STATUS_ID_NEW"),
+				Vulnerabilities: []*ocsf.Vulnerability{
+					{
+						AffectedCode: []*ocsf.AffectedCode{
+							{
+								File: &ocsf.File{
+									Name: "go-dvwa/vulnerable/system.go",
+									Path: ptr.Ptr("file://go-dvwa/vulnerable/system.go"),
+								},
+								StartLine: ptr.Ptr(int32(9)),
+								EndLine:   ptr.Ptr(int32(9)),
+							},
+						},
+						Desc:            ptr.Ptr("Detected non-static command inside Command. Audit the input to 'exec.Command'. If unverified user data can reach this call site, this is a code injection vulnerability. A malicious actor can inject a malicious script to execute arbitrary code."),
+						FirstSeenTime:   ptr.Ptr(now.Unix()),
+						FirstSeenTimeDt: timestamppb.New(now),
+						FixAvailable:    ptr.Ptr(false),
+						IsFixAvailable:  ptr.Ptr(false),
+						LastSeenTime:    ptr.Ptr(now.Unix()),
+						LastSeenTimeDt:  timestamppb.New(now),
+						Severity:        ptr.Ptr("SEVERITY_ID_MEDIUM"),
+						Title:           ptr.Ptr("go.lang.security.audit.dangerous-exec-command.dangerous-exec-command"),
+						VendorName:      ptr.Ptr("Semgrep OSS"),
+					},
+				},
+			},
+		}
+		transformer, err := sariftransformer.NewTransformer(&sarifOutput, "", sariftransformer.TargetTypeRepository, clock, MockUUIDProvider{FixedUUID: fixedUUID})
+		require.NoError(t, err)
+		actualIssues, err := transformer.ToOCSF(context.Background())
+		require.NoError(t, err)
+		require.Equal(t, len(actualIssues), len(expectedIssues))
+		// handle datasource differently see https://github.com/golang/protobuf/issues/1121
+		for i, e := range expectedIssues {
+			var expectedDataSource, actualDatasource ocsffindinginfo.DataSource
+			require.Equal(t, len(e.FindingInfo.DataSources), len(actualIssues[i].FindingInfo.DataSources))
+
+			for j, d := range e.GetFindingInfo().DataSources {
+				protojson.Unmarshal([]byte(d), &expectedDataSource)
+				protojson.Unmarshal([]byte(actualIssues[i].FindingInfo.DataSources[j]), &actualDatasource)
+				require.EqualExportedValues(t, &expectedDataSource, &actualDatasource)
+			}
+			expectedIssues[i].FindingInfo.DataSources = nil
+			actualIssues[i].FindingInfo.DataSources = nil
+		}
+		require.EqualExportedValues(t, expectedIssues, actualIssues)
+
+	})
+	t.Run("trivy testcase", func(t *testing.T) {
+		exampleOutput, err := os.ReadFile("./testdata/trivy_output.json")
```

Comment:
@andream16 can you add the CVE ID in the expected finding for trivy please?

Reply:
Can do but i'd prefare to do it separately as I need to get a valid trivy output and will take me some time

Reply:
Done